__tests__/intergration.test.js

```diff
@@ -190,22 +190,50 @@ describe("GET /api/articles/:article_id/comments",()=>{
     })
 })
 
-//POST COMMENT ON AN ARTICLE==========================
-
-xdescribe("POST /api/articles/:article_id/comments",()=>{
-    test("POST: 201 - Posted a new comment to the article",()=>{
-        const testComment={
-            body: "This article would actually be really informative if it was re-written by my 8 year old.",
-            votes: 23,
-            author: "Negs"
+//POST A COMMENT ON AN ARTICLE========================
+describe("POST /api/articles/:article_id/comments",()=>{
+    test("POST: 201 - Posted a comment on an article",()=>{
+        const testComment={username: "butter_bridge",
+            body:"Lorum Ipsum..."
         }
         return request(app)
-        .post("/api/articles/1/comments")
+        .post("/api/articles/3/comments")
         .send(testComment)
         .expect(201)
-        .then(()=>{
+        .then(({body:{newComment}})=>{
+            expect(newComment.comment_id).toBe(19)
+            expect(newComment.body).toBe("Lorum Ipsum...")
+            expect(newComment.article_id).toBe(3)
+            expect(newComment.votes).toBe(0)
+            expect(newComment.author).toBe("butter_bridge")
+            expect(typeof newComment.created_at).toBe("string")
+        })
+    })
+    test("Error: 404 - Cannot Find That Article ID",()=>{
+        const testComment={username: "butter_bridge",
+            body:"Lorum Ipsum..."
+        }
+        return request(app)
+        .post("/api/articles/999/comments")
+        .send(testComment)
+        .expect(404)
+        .then(({body:{msg}})=>{
+            expect(msg).toBe("Error 404 - Article Not Found")
+        })
 
+    })
+    test("Error: 400 - Bad request, ID is not correct data type",()=>{
+        const testComment={username: "butter_bridge",
+            body:"Lorum Ipsum..."
+        }
+        return request(app)
+        .post("/api/articles/string/comments")
+        .send(testComment)
+        .expect(400)
+        .then(({body:{msg}})=>{
+            expect(msg).toBe("Error 400 - Bad Request Given")
         })
+
     })
```

Comment:
You are missing some tests here
- what if the request body does not have a `username` or a `body` key?
- what if the username does not exist?
- can i still post a comment when there are other properties in the request body?

Reply:
Oh, I skipped over these because I convinced myself this isn't an issue because if a username doesn't exist then a user does not exist and therefore they cannot type a comment and similarly with body, a user cannot post if they've not typed anything. Anyways will fix this!